pyanaconda/modules/boss/install_manager/install_manager.py

```diff
@@ -115,17 +113,17 @@ def start_installation(self):
     def _collect_tasks(self):
         self._tasks.clear()
 
-        if not self._modules:
+        if not self._module_observers:
             log.error("Starting installation without available modules.")
 
-        for module_service in self._modules:
-            # FIXME: This is just a temporary solution.
-            module_object = DBus.get_proxy(module_service, auto_object_path(module_service))
+        for observer in self._module_observers:
+            if not observer.is_service_available:
```

Comment:
Maybe we could catch the DBusObserverError instead ? Also I think it could be useful to log if a module observer we try to access is not actually usable.

Reply:
I agree with the logging.
As for catching the exception, I think the approach of - check availability and raise exception (without catching/handling) when calling observer.proxy methods - seems OK to me here.

Reply:
The check is here so we can test the StartInstallation method even with some failed modules. It shouldn't be here once we will start to use it for real. I will add a log there and a comment.